packages/kbn-test/src/kbn_client/kbn_client_requester.ts

```diff
@@ -116,61 +116,64 @@ export class KbnClientRequester {
-    const description = options.description || `${options.method} ${url}`;
+    const redacted = redactUrl(url);
     let attempt = 0;
     const maxAttempts = options.retries ?? DEFAULT_MAX_ATTEMPTS;
-
     while (true) {
       attempt += 1;
-
       try {
-        const response = await Axios.request({
-          method: options.method,
-          url,
-          data: options.body,
-          params: options.query,
-          headers: {
-            ...options.headers,
-            'kbn-xsrf': 'kbn-client',
-            'x-elastic-internal-origin': 'kbn-client',
-          },
-          httpsAgent: this.httpsAgent,
-          responseType: options.responseType,
-          // work around https://github.com/axios/axios/issues/2791
-          transformResponse: options.responseType === 'text' ? [(x) => x] : undefined,
-          maxContentLength: 30000000,
-          maxBodyLength: 30000000,
-          paramsSerializer: (params) => Qs.stringify(params),
-        });
-
-        return response;
+        this.log.info(`Requesting url (redacted): [${redacted}]`);
+        return await Axios.request(buildRequest(url, this.httpsAgent, options));
       } catch (error) {
-        const conflictOnGet = isConcliftOnGetError(error);
-        const requestedRetries = options.retries !== undefined;
-        const failedToGetResponse = isAxiosRequestError(error);
-
-        if (isIgnorableError(error, options.ignoreErrors)) {
-          return error.response;
-        }
-
-        let errorMessage;
-        if (conflictOnGet) {
-          errorMessage = `Conflict on GET (path=${options.path}, attempt=${attempt}/${maxAttempts})`;
-          this.log.error(errorMessage);
-        } else if (requestedRetries || failedToGetResponse) {
-          errorMessage = `[${description}] request failed (attempt=${attempt}/${maxAttempts}): ${error.message}`;
-          this.log.error(errorMessage);
-        } else {
-          throw error;
-        }
-
+        if (isIgnorableError(error, options.ignoreErrors)) return error.response;
         if (attempt < maxAttempts) {
           await delay(1000 * attempt);
           continue;
         }
-
-        throw new KbnClientRequesterError(`${errorMessage} -- and ran out of retries`, error);
+        throw new KbnClientRequesterError(`${errMsg(error)} -- and ran out of retries`, error);
+
+        function errMsg(_: any): string {
+          const requestedRetries: boolean = options.retries !== undefined;
+          const failedToGetResponse: boolean = isAxiosRequestError(_);
+          const result = isConcliftOnGetError(_)
+            ? `Conflict on GET (path=${options.path}, attempt=${attempt}/${maxAttempts})`
+            : requestedRetries || failedToGetResponse
+            ? `[${
+                options.description || `${options.method} - ${redacted}`
+              }] request failed (attempt=${attempt}/${maxAttempts}): ${_?.code}`
+            : '';
+          if (result === '') throw _;
```

Comment:
I only added it to match the `request` methods expected shape at runtime, for the tests that are failing.  
Let's see if this fixes everything.

Reply:
This fixed it.